### ISSUE TITLE
Don't report error missing initializer in const declaration in for...of and for...in statement

### ISSUE THREAD

**Issue by singalhimanshu** [quick-lint/quick-lint-js#380]
The following code currently reports `error: missing initializer in const declaration [E205]`, it should not report any errors

for-of:
```js
const array1 = ['a', 'b', 'c'];

for (const element of array1) {
  console.log(element);
}
```

for-in:
```js
const array1 = ['a', 'b', 'c'];

for (const element in array1) {
  console.log(element);
}
```

**Comment by singalhimanshu** [comment]
This bug was introduced in commit c7247245a14452d314ac26f0197c89301291333b.

**Comment by strager** [comment]
The following are cases I thought of related and semi-related to this task. The cases marked with `!!` are where quick-lint-js is doing the wrong thing (false positive or false negative):
```js
let xs = [];
function cond() {}
function update() {}

for (const x = 3; cond(); update()) {}        //
for (const x = 3, y = 4; cond(); update()) {} //
for (const x = 3 of xs) {}                    // !! should error, but doesn't
for (const x of xs) {}                        // !! shouldn't error, but does
for (const x, y of xs) {}                     // should error
for (let x, y of xs) {}                       // !! should error, but doesn't
for (const x = 3 in xs) {}                    // !! should error, but doesn't
for (const x in xs) {}                        // !! shouldn't error, but does
for (const x, y in xs) {}                     // should error
for (let x, y in xs) {}                       // !! should error, but doesn't
for (const x; cond(); update()) {}            // should error
for (const x, y = 3; cond(); update()) {}     // should error
for (const x = 3, y; cond(); update()) {}     // should error
```

**Comment by singalhimanshu** [comment]
I think these are the 3 different tasks that I'm able to segregate:
- [ ] `for of` and `for in` should not have assignment
```js
let xs = [];
for (const x = 3 in xs) {}
for (const x = 3 of xs) {}
```
- [ ] const in `for of` and `for in`  should not error
```js
let xs = [];
for (const x of xs) {}
for (const x in xs) {}
```
- [ ] Multiple variables are not allowed in `for of` and `for in`
```js
let xs = [];
for (let x, y of xs) {}
for (let x, y in xs) {}
```

**Comment by strager** [comment]
> I think these are the 3 different tasks that I'm able to segregate:

Your split makes sense.

The second case matters the most right now, since it's a false positive.

**Comment by singalhimanshu** [comment]
Fix for the second case is in PR: #421
